File: GridPatternExample/src/main/java/org/akka/essentials/grid/controller/RegisterRemoteWorkerActor.java
package org.akka.essentials.grid.controller;

import org.akka.essentials.grid.StartWorker;

import akka.actor.ActorRef;
import akka.actor.UntypedActor;
import akka.event.Logging;
import akka.event.LoggingAdapter;

public class RegisterRemoteWorkerActor extends UntypedActor {

	LoggingAdapter log = Logging.getLogger(getContext().system(), this);
	private ActorRef jobControllerActor;

	@Override
	public void onReceive(Object message) throws Exception {
		if (message instanceof StartWorker) {
			log.info("Recieved Registration Info from " + message.toString());
			jobControllerActor.tell(message);
		} else
			log.error("Wrong message type recieved");
	}

	public RegisterRemoteWorkerActor(ActorRef inJobControllerActor) {
		jobControllerActor = inJobControllerActor;
	}

}

File: GridPatternExample/src/main/scala/org/akka/essentials/grid/controller/RegisterRemoteWorkerActor.scala
package org.akka.essentials.grid.controller
import akka.actor.ActorRef
import akka.actor.ActorLogging
import akka.actor.Actor

class RegisterRemoteWorkerActor(val jobControllerActor: ActorRef) extends Actor with ActorLogging {
  def receive: Receive = {
    case message: StartWorker =>
      log.info("Received Registration Info from " + message.toString());
      jobControllerActor.tell(message);
    case _ =>
      log.error("Wrong message type received");
  }
}